pkg/operator/ceph/cluster/mon/config_test.go

```diff
@@ -100,8 +101,23 @@ func TestCreateClusterSecrets(t *testing.T) {
 	}
 	_, err = clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
 	assert.NoError(t, err)
-	info, _, _, err = CreateOrLoadClusterInfo(context, ctx, namespace, ownerInfo)
+	info, _, _, err = CreateOrLoadClusterInfo(context, ctx, namespace, ownerInfo, v1.ClusterSpec{})
 	assert.NoError(t, err)
 	assert.Equal(t, "testid", info.CephCred.Username)
 	assert.Equal(t, "testkey", info.CephCred.Secret)
+
+	// test annotation
+	_, _, _, err = CreateOrLoadClusterInfo(context, ctx, namespace, ownerInfo, v1.ClusterSpec{Annotations: v1.AnnotationsSpec{v1.KeyClusterMetadata: {"key": "value"}}})
+
+	assert.NoError(t, err)
+	secret, err = clientset.CoreV1().Secrets(namespace).Get(ctx, "rook-ceph-mon", metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, map[string]string{"key": "value"}, secret.Annotations)
+
+	_, _, _, err = CreateOrLoadClusterInfo(context, ctx, namespace, ownerInfo, v1.ClusterSpec{Annotations: v1.AnnotationsSpec{v1.KeyClusterMetadata: {"key1": "value1"}}})
+
+	assert.NoError(t, err)
+	secret, err = clientset.CoreV1().Secrets(namespace).Get(ctx, "rook-ceph-mon", metav1.GetOptions{})
+	assert.NoError(t, err)
+	assert.Equal(t, map[string]string{"key": "value", "key1": "value1"}, secret.Annotations)
```

Comment:
@travisn notice this unit test and other unit tests [here](https://github.com/rook/rook/pull/9547/files#diff-1b9625ce81dfcf6565ac183a2a3daae684ef64a36e4b39b2dbef0f562ec746ecR138).
 
I'll explain with an example what diff I noticed

1. suppose with created ceph cluster with:
```yaml
clusterMetadata:
      key: value
```
 All will have annotation `key:value` .

2. when updating with 
```yaml
clusterMetadata:
      key: value
      key1: value1
```
All will have annotation `key: value` and `key1:value1`.

 3. But when we update the cephCluster
```yaml
clusterMetadata:
      key1: value1
```
 `rook-ceph-mon` will have both old one `key:value` and new one`key1:value1`. Other will have `key1:value1`(this is what expected).

Reply:
Ah, this is because the rook-ceph-mon secret is retrieved and then updated, instead of a new secret being generated by the operator. So we would need to reset the annotations back to empty, then reapply the annotations in that case.